tensorflow/core/ops/batchnorm_training_op.cc

```diff
@@ -0,0 +1,80 @@
+/* Copyright 2015 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/framework/op.h"
+
+namespace tensorflow {
+
+REGISTER_OP("BatchNormalizeTraining")
+    .Input("input: T")
+    .Input("scale: T")
+    .Input("bias: T")
+    .Input("running_mean: Ref(T)")
```

Comment:
We shouldn't have this op modify the running_mean and running_inv_var directly. A few problems: 

1. making distributed training difficult, since all replicas needs to directly access the running mean; 
2. synchronized training difficult, since multiple replicas do all the update without proper locking.